src/api/objects/ObjectAPI.js

```diff
@@ -555,27 +555,39 @@ export default class ObjectAPI {
   async getTelemetryPath(identifier, telemetryIdentifier) {
     const objectDetails = await this.get(identifier);
     const telemetryPath = [];
-    if (objectDetails.composition && !['folder'].includes(objectDetails.type)) {
-      let sourceTelemetry = objectDetails.composition[0];
+    if (['folder'].includes(objectDetails.type)) {
+      return telemetryPath;
+    }
+
+    let sourceTelemetry = null;
+    let isParentSameAsChild = true;
+    Object.keys(identifier).forEach((key) => {
```

Comment:
Let's use [identifierEquals()](https://github.com/nasa/openmct/blob/f3b819a7867ab8089be7c50de54bebe6d2e0c5a5/src/api/objects/object-utils.js#L150) for this

Reply:
ooh, good call